src/client/jupyter/kernel-manager.ts

```diff
@@ -78,26 +78,25 @@ export class KernelManagerImpl extends EventEmitter {
         });
     }
 
-    public startKernelFor(language: string): Promise<Kernel> {
-        return this.getKernelSpecFor(language).then(kernelSpec => {
-            return this.startKernel(kernelSpec, language);
-        });
+    public async startKernelFor(language: string): Promise<Kernel> {
+        const kernelSpec = await this.getKernelSpecFor(language);
+        return this.startKernel(kernelSpec, language);
     }
 
     public startExistingKernel(language: string, connection, connectionFile): Promise<Kernel> {
         throw new Error('Start Existing Kernel not implemented');
     }
 
-    public startKernel(kernelSpec: KernelspecMetadata, language: string): Promise<Kernel> {
+    public async startKernel(kernelSpec: KernelspecMetadata, language: string): Promise<Kernel> {
```

Comment:
Sorry still away on holidays. I can see that you have also attempted to clean up some of the code by replacing promises with async, in totally unrelated code. Wouldn't it be better to introduce such clean up in a separate PR?